### PR TITLE
Remove blsToExecutionChange field and related logics

### PR DESCRIPTION
**Pull request by qqqeck**
Remove field BlsToExecutionChange from block (Capella~).

Note: About TestPublishBlock(V2) - wrong version header, as far as I see the original code randomly selected Bellatrix and Capella to test the wrong header version, which doesn't work in this PR because the only difference in the block between the two forks are BlsToExecutionChange. As far as I see working around this by comparing Deneb and Bellatrix is ok because the purpose of the test look like to see if it returns the said error message when content is not decodable. TestPublishBlockSSZ is ok because execution payload ssz changed a little bit in Bellatrix->Capella.

### CODE REVIEW

File: beacon-chain/blockchain/head_test.go [hunk 1]
```diff
@@ -461,9 +459,7 @@ func TestSaveOrphanedAtts_CanFilter(t *testing.T) {
 
 	require.NoError(t, service.saveOrphanedOperations(ctx, r2, r4))
 	require.Equal(t, 1, service.cfg.AttPool.AggregatedAttestationCount())
-	pending, err := service.cfg.BLSToExecPool.PendingBLSToExecChanges()
 	require.NoError(t, err)
```
Comment (syjn99) [inline beacon-chain/blockchain/head_test.go]: This line can be also removed.

**Review by syjn99 — COMMENTED** [review]
```go
// BlockGenConfig is used to define the requested conditions
// for block generation.
type BlockGenConfig struct {
	NumProposerSlashings uint64
	NumAttesterSlashings uint64
	NumAttestations      uint64
	NumDeposits          uint64
	NumVoluntaryExits    uint64

	NumTransactions       uint64 // Only for post Bellatrix blocks
	NumBLSChanges         uint64 // Only for post Capella blocks
	NumWithdrawals        uint64
	NumDepositRequests    uint64 // Only for post Electra blocks
	NumWithdrawalRequests uint64 // Only for post Electra blocks
}
```

In `testing/util/block.go`, `NumBLSChanges` can also be removed.

File: consensus-types/blocks/kzg.go [hunk 2]
```diff
@@ -206,17 +206,6 @@ func topLevelRoots(body interfaces.ReadOnlyBeaconBlockBody) ([][]byte, error) {
...
-	if err != nil {
-		return nil, err
-	}
-	root, err = ssz.MerkleizeListSSZ(bls, params.BeaconConfig().MaxBlsToExecutionChanges)
-	if err != nil {
-		return nil, err
-	}
-	copy(layer[9], root[:])
-
 	// KZG commitments is not needed. skip index 10
```
Comment (syjn99) [inline consensus-types/blocks/kzg.go]:
```suggestion
	// KZG commitments is not needed. skip index 9
```
Just a nit

**Review by syjn99 — APPROVED** [review]
LGTM

File: beacon-chain/blockchain/head_test.go (hunk 1 above)
Comment (qqqeck) [reply beacon-chain/blockchain/head_test.go]:
```suggestion
```

**Review by gts2030 — APPROVED** [review]

**Pull request merged**